modules/sdk/ant-bnd/src/main/java/com/liferay/ant/bnd/resource/bundle/AggregateResourceBundleLoaderAnalyzerPlugin.java

```diff
@@ -0,0 +1,134 @@
+
+import aQute.bnd.header.Attrs;
+import aQute.bnd.header.Parameters;
+import aQute.bnd.osgi.Analyzer;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.service.AnalyzerPlugin;
+
+import aQute.libg.filters.AndFilter;
+import aQute.libg.filters.LiteralFilter;
+import aQute.libg.filters.NotFilter;
+import aQute.libg.filters.SimpleFilter;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Gregory Amerson
+ */
+public class AggregateResourceBundleLoaderAnalyzerPlugin
+	implements AnalyzerPlugin {
+
+	@Override
+	public boolean analyzeJar(Analyzer analyzer) throws Exception {
+		Parameters aggregateResourceBundlesParameters = new SortedParameters(
+			analyzer.getProperty("-aggregateresourcebundles"));
+
+		if (aggregateResourceBundlesParameters.size() == 0) {
+			return false;
+		}
+
+		Parameters provideCapabilityHeaders = new SortedParameters(
+			analyzer.getProperty(Constants.PROVIDE_CAPABILITY));
+
+		Parameters requireCapabilityHeaders = new SortedParameters(
+			analyzer.getProperty(Constants.REQUIRE_CAPABILITY));
+
+		List<String> aggregateKeys = new ArrayList<>(
+			aggregateResourceBundlesParameters.keySet());
+
+		Collections.sort(aggregateKeys);
+
+		for (String aggregateResourceBundlesParameter : aggregateKeys) {
+			Attrs requireAttrs = new Attrs();
+
+			requireAttrs.put(
+				"filter:",
+				new SimpleFilter(
+					"bundle.symbolic.name",
+					aggregateResourceBundlesParameter).toString());
+
+			requireCapabilityHeaders.add(
+				"liferay.resource.bundle", requireAttrs);
+		}
+
+		analyzer.setProperty(
+			Constants.REQUIRE_CAPABILITY, requireCapabilityHeaders.toString());
+
+		StringBuilder aggregateValue = new StringBuilder();
+
+		AndFilter aggregateSelfFilter = new AndFilter();
+
+		aggregateSelfFilter.addChild(
+			new SimpleFilter("bundle.symbolic.name", analyzer.getBsn()));
+		aggregateSelfFilter.addChild(
+			new NotFilter(new LiteralFilter("(aggregate=true)")));
+		aggregateSelfFilter.append(aggregateValue);
+
+		for (String aggregateResourceBundlesParameter : aggregateKeys) {
+			aggregateValue.append(",");
+
+			new SimpleFilter(
+				"bundle.symbolic.name",
+				aggregateResourceBundlesParameter).append(aggregateValue);
+		}
+
+		Attrs provideAttrs = new Attrs();
+
+		provideAttrs.put(
+			"resource.bundle.aggregate:String", aggregateValue.toString());
+		provideAttrs.put("bundle.symbolic.name", analyzer.getBsn());
+		provideAttrs.put("resource.bundle.base.name", "content.Language");
+		provideAttrs.put("service.ranking:Long", "1");
+		provideAttrs.put("aggregate", "true");
+
+		String servletContextName = analyzer.getProperty("Web-ContextPath");
+
+		if (servletContextName != null) {
+			try {
+				Path servletContextPath = Paths.get(servletContextName);
+
+				servletContextName = servletContextPath.subpath(
+					0, 1).toString();
+			}
+			catch (Exception e) {
+
+				// ignore
```

Comment:
Its a Paths.get() exception.  So if the `Web-Contextpath` can't get parsed by `Paths.get()` then it just doesn't correctly encode the servletContextName.  I guess we shoudl fail the plugin or fail the build maybe?  So the developer knows to fix their `Web-ContextPath`?